src/bitmap/store/bitmap_store.rs

```diff
@@ -237,6 +252,16 @@ impl BitmapStore {
     }
 }
 
+// this can be done in 3 instructions on x86-64 with bmi2 with: tzcnt(pdep(1 << rank, value))
+// if n > value.count_ones() this method returns 0
+fn select(mut value: u64, n: u64) -> u64 {
+    // reset n of the least significant bits
+    for _ in 0..n {
+        value &= value - 1;
+    }
+    value.trailing_zeros() as u64
+}
```

Comment:
Yeah, that's indeed what should have been done! I badly understood the `select` method, I thought it was a function that was returning whether a given bit was on or off. Sorry.